src/setup.ts

```diff
@@ -21,8 +21,9 @@ const mongo: Mongo = isReplSet
   ? new MongoMemoryReplSet(mongoMemoryServerOptions)
   : new MongoMemoryServer(mongoMemoryServerOptions);
 
-module.exports = async (config: JestEnvironmentConfig['projectConfig']) => {
-  const globalConfigPath = join(config.rootDir, 'globalConfig.json');
+module.exports = async (config: JestEnvironmentConfig['globalConfig']) => {
+  const projects = config.projects.length ? config.projects : [config.rootDir];
+  const globalConfigPaths = projects.map(project => join(project, 'globalConfig.json'));
```

Comment:
`config.projects` can also be a list of sub-config. See https://jestjs.io/docs/configuration#projects-arraystring--projectconfig
We should probably have something like that:
```suggestion
  const projects = config.projects.length ? config.projects : [config];
  const globalConfigPaths = projects.map(project.rootDir => join(project, 'globalConfig.json'));
```
What do you think?

Reply:
@Grmiade looks like you are correct, if I set up a jest config using a combination of "string" projects and objects with project configuration, the `setup` function is called with that same array. I can definitely fix it for that, but I'm struggling with the type definition to use here. The Jest source shows this function being called with an object of type "GlobalConfig"
https://github.com/facebook/jest/blob/aae0d3ac3498ece4673ec29277dc040234f9f643/packages/jest-core/src/runGlobalHook.ts#L58

but that type only specifies `projects` as being an array of strings, which seems incorrect:
https://github.com/facebook/jest/blob/aae0d3ac3498ece4673ec29277dc040234f9f643/packages/jest-types/src/Config.ts#L392

it's defined correctly in "InitialOptions" but I don't think that represents the rest of what's passed in there correctly either...
https://github.com/facebook/jest/blob/aae0d3ac3498ece4673ec29277dc040234f9f643/packages/jest-types/src/Config.ts#L280

Reply:
disregard all that, updated the PR to just use the `globalConfig` rootDir field (which is always set and works for both use-cases as well.)